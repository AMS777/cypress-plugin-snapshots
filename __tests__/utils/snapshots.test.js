const { subjectToSnapshot } = require('../../utils/snapshots');
const configModule = require('../../config');

jest.mock("../../config.js");

jest.spyOn(configModule, 'initConfig')
  .mockImplementation((config) => config);

global.Cypress = {
  env: () => {},
  config: () => {},
  Commands: {
    add: jest.fn(),
  },
};

global.cy = {};

describe('utils/snapshot', () => {
  describe('keepKeysFromExpected', () => {
    const expected = {
      field: "value",
      items: [{
          name: "item1",
          "items": [{
            name: "subitem1"
          }]
        },
        {
          name: "item2"
        },
        {
          name: "item3"
        },
      ]
    };

    const actual = {
      field: "value",
      extra: "extra",
      items: [{
          name: "item1",
          "items": [{
            name: "subitem1",
            extra: 'extra1'
          }, {
            name: "subitem2",
            extra: 'extra2'
          }]
        },
        {
          name: "item2"
        },
        {
          name: "item3"
        },
        {
          name: "item4"
        },
      ]
    };

    it('ignoreExtraFields: false, ignoreExtraArrayItems: false', () => {
      const config = {
        ignoreExtraFields: false,
        ignoreExtraArrayItems: false,
      };
      jest.spyOn(configModule, 'getConfig').mockImplementation(() => config);

      const {
        keepKeysFromExpected
      } = require('../../utils/snapshots');

      const result = keepKeysFromExpected(actual, expected);
      expect(result).toMatchSnapshot();
    });

    it('ignoreExtraFields: true, ignoreExtraArrayItems: false', () => {
      const config = {
        ignoreExtraFields: true,
        ignoreExtraArrayItems: false,
      };
      jest.spyOn(configModule, 'getConfig').mockImplementation(() => config);

      const {
        keepKeysFromExpected
      } = require('../../utils/snapshots');
      const result = keepKeysFromExpected(actual, expected);
      expect(result).toMatchSnapshot();
    });

    it('ignoreExtraFields: true, ignoreExtraArrayItems: true', () => {
      const config = {
        ignoreExtraFields: true,
        ignoreExtraArrayItems: true,
      };
      jest.spyOn(configModule, 'getConfig').mockImplementation(() => config);
      const {
        keepKeysFromExpected
      } = require('../../utils/snapshots');

      const result = keepKeysFromExpected(actual, expected);
      expect(result).toMatchSnapshot();
    });

    it('ignoreExtraFields: false, ignoreExtraArrayItems: true', () => {
      const config = {
        ignoreExtraFields: false,
        ignoreExtraArrayItems: true,
      };
      jest.spyOn(configModule, 'getConfig').mockImplementation(() => config);

      const {
        keepKeysFromExpected
      } = require('../../utils/snapshots');

      const result = keepKeysFromExpected(actual, expected);
      expect(result).toMatchSnapshot();
    });
  });

  describe('subjectToSnapshot', () => {
    it('normalizes', () => {
      const normalized = subjectToSnapshot({
        foo: 'bar',
        bar: 'foo',
        sub: {
          foo: 'bar',
          bar: 'foo',
          asubsub: {
            foo: 'bar',
            bar: 'foo',
          }
        }
      }, true);

      expect(normalized).toMatchSnapshot();
    });
  });
});
