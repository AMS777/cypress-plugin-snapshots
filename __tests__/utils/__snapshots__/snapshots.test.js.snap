// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`utils/snapshot keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: false 1`] = `
Object {
  "extra": "extra",
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "name": "subitem2",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": "item4",
    },
  ],
}
`;

exports[`utils/snapshot keepKeysFromExpected ignoreExtraFields: false, ignoreExtraArrayItems: true 1`] = `
Object {
  "extra": "extra",
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "extra": "extra1",
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
  ],
}
`;

exports[`utils/snapshot keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: false 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
        Object {
          "extra": "extra2",
          "name": "subitem2",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
    Object {
      "name": "item4",
    },
  ],
}
`;

exports[`utils/snapshot keepKeysFromExpected ignoreExtraFields: true, ignoreExtraArrayItems: true 1`] = `
Object {
  "field": "value",
  "items": Array [
    Object {
      "items": Array [
        Object {
          "name": "subitem1",
        },
      ],
      "name": "item1",
    },
    Object {
      "name": "item2",
    },
    Object {
      "name": "item3",
    },
  ],
}
`;

exports[`utils/snapshot subjectToSnapshot normalizes 1`] = `
Object {
  "bar": "foo",
  "foo": "bar",
  "sub": Object {
    "asubsub": Object {
      "bar": "foo",
      "foo": "bar",
    },
    "bar": "foo",
    "foo": "bar",
  },
}
`;
